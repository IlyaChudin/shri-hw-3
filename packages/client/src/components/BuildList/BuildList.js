import React from "react";
import cn from "../../classname";
import BuildCard from "../BuildCard";
import Button from "../Button";

const buildList = cn("build-list");

function BuildList(props) {
  const { builds, showMore, onShowMoreClick } = props;
  return (
    <div className={buildList()}>
      {builds.map(build => (
        <BuildCard
          key={build.id}
          href={`/build/${build.id}`}
          status={build.status}
          mix={buildList("item")}
          number={build.buildNumber}
          title={build.commitMessage}
          commitBranch={build.branchName}
          commitHash={build.commitHash}
          user={build.authorName}
          date={build.start}
          duration={build.duration}
        />
      ))}
      {showMore && <Button text="Show more" size="m" mix={buildList("button")} onClick={onShowMoreClick} />}
    </div>
  );
}

export default BuildList;
