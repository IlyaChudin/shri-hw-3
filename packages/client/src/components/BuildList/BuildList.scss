@import "../../mixins";

.build-list {
  &__item {
    margin-bottom: 8px;
  }

  &__button {
    min-height: 28px;
    line-height: 28px;

    & .button__text {
      margin-left: 13px;
      margin-right: 13px;
    }

    @include mobile {
      width: 100%;
      min-height: 36px;
      line-height: 36px;

      & .button__text {
        margin-left: 20px;
        margin-right: 20px;
      }
    }
  }
}
