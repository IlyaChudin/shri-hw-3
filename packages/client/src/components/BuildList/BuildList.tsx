import React from "react";
import { BuildModel } from "@shri-ci/types";
import cn from "../../classname";
import BuildCard from "../BuildCard";
import Button from "../Button";
import Loading from "../Loading";

const buildList = cn("build-list");

interface BuildListProps {
  builds: BuildModel[];
  showMore: boolean;
  isLoading: boolean;
  onShowMoreClick: (event: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void;
}

const BuildList: React.FC<BuildListProps> = ({ builds, showMore, isLoading, onShowMoreClick }) => {
  return (
    <div data-testid="build-list" className={buildList()}>
      {builds.map(build => (
        <BuildCard key={build.id} href={`/build/${build.id}`} mix={buildList("item")} {...build} />
      ))}
      {isLoading && <Loading />}
      {showMore && !isLoading && (
        <Button text="Show more" size="m" mix={buildList("button")} onClick={onShowMoreClick} />
      )}
    </div>
  );
};

export default BuildList;
