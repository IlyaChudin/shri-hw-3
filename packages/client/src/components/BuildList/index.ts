import BuildList from "./BuildList";
import "./BuildList.scss";

export default BuildList;
