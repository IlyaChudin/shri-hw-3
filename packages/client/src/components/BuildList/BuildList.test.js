import React from "react";
import renderer from "react-test-renderer";
import { MemoryRouter } from "react-router-dom";
import BuildList from "./BuildList";

const builds = [
  {
    id: "f7981650-50c6-4eb2-a357-f112910a3eb7",
    configurationId: "952b0cf1-948a-4f95-b6ce-ee225d36c24c",
    buildNumber: 37,
    commitMessage: "test update",
    commitHash: "0cab04f6b0894e775e66224469fde0309f8eb284",
    branchName: "master",
    authorName: "Ilya Chudin",
    status: "Waiting"
  },
  {
    id: "01d78ffb-dae2-4bb9-9457-233fc8700c7e",
    configurationId: "952b0cf1-948a-4f95-b6ce-ee225d36c24c",
    buildNumber: 36,
    commitMessage: "test update",
    commitHash: "0cab04f6b0894e775e66224469fde0309f8eb284",
    branchName: "master",
    authorName: "Ilya Chudin",
    status: "Success",
    start: "2020-04-04T08:32:53.442",
    duration: 2375
  }
];

describe("BuildList component", () => {
  it.each([
    [true, true],
    [true, false],
    [false, true],
    [false, false]
  ])("should renders correctly", (showMore, isLoading) => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <BuildList builds={builds} showMore={showMore} isLoading={isLoading} />
        </MemoryRouter>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
