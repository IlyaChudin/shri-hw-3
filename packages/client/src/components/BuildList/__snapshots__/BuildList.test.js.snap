// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BuildList component should renders correctly 1`] = `
<div
  className="build-list"
  data-testid="build-list"
>
  <div
    className="build-card build-card_view_default build-card_type_pending build-list__item"
    onClick={[Function]}
  >
    <div
      className="build-card__status"
    />
    <div
      className="build-card__container"
    >
      <div>
        <div
          className="build-card__header"
        >
          <div
            className="build-card__number"
          >
            #
            37
          </div>
          <div
            className="build-card__title"
          >
            test update
          </div>
        </div>
        <div
          className="build-card__info"
        >
          <div
            className="icon-plus build-card__commit"
          >
            <div
              className="icon icon_type_commit icon_size_xs"
            />
            <span
              className="icon-plus__text icon-plus__text_type_primary"
            >
              master
            </span>
            <span
              className="icon-plus__text icon-plus__text_type_secondary"
            >
              0cab04f
            </span>
          </div>
          <div
            className="icon-plus build-card__user"
          >
            <div
              className="icon icon_type_user icon_size_xs"
            />
            <span
              className="icon-plus__text icon-plus__text_type_primary"
            >
              Ilya Chudin
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="build-card build-card_view_default build-card_type_success build-list__item"
    onClick={[Function]}
  >
    <div
      className="build-card__status"
    />
    <div
      className="build-card__container"
    >
      <div>
        <div
          className="build-card__header"
        >
          <div
            className="build-card__number"
          >
            #
            36
          </div>
          <div
            className="build-card__title"
          >
            test update
          </div>
        </div>
        <div
          className="build-card__info"
        >
          <div
            className="icon-plus build-card__commit"
          >
            <div
              className="icon icon_type_commit icon_size_xs"
            />
            <span
              className="icon-plus__text icon-plus__text_type_primary"
            >
              master
            </span>
            <span
              className="icon-plus__text icon-plus__text_type_secondary"
            >
              0cab04f
            </span>
          </div>
          <div
            className="icon-plus build-card__user"
          >
            <div
              className="icon icon_type_user icon_size_xs"
            />
            <span
              className="icon-plus__text icon-plus__text_type_primary"
            >
              Ilya Chudin
            </span>
          </div>
        </div>
      </div>
      <div
        className="build-card__time"
      >
        <div
          className="icon-plus build-card__date"
        >
          <div
            className="icon icon_type_calendar icon_size_xs"
          />
          <span
            className="icon-plus__text icon-plus__text_type_secondary"
          >
            4 апр. 08:32
          </span>
        </div>
        <div
          className="icon-plus build-card__duration"
        >
          <div
            className="icon icon_type_stopwatch icon_size_xs"
          />
          <span
            className="icon-plus__text icon-plus__text_type_secondary"
          >
            39 м 35 сек
          </span>
        </div>
      </div>
    </div>
  </div>
  <div
    className="loading"
  >
    <div
      className="loading__container"
    >
      <div
        className="loading__bar"
      />
      <div
        className="loading__bar"
      />
      <div
        className="loading__bar"
      />
    </div>
  </div>
</div>
`;

exports[`BuildList component should renders correctly 2`] = `
<div
  className="build-list"
  data-testid="build-list"
>
  <div
    className="build-card build-card_view_default build-card_type_pending build-list__item"
    onClick={[Function]}
  >
    <div
      className="build-card__status"
    />
    <div
      className="build-card__container"
    >
      <div>
        <div
          className="build-card__header"
        >
          <div
            className="build-card__number"
          >
            #
            37
          </div>
          <div
            className="build-card__title"
          >
            test update
          </div>
        </div>
        <div
          className="build-card__info"
        >
          <div
            className="icon-plus build-card__commit"
          >
            <div
              className="icon icon_type_commit icon_size_xs"
            />
            <span
              className="icon-plus__text icon-plus__text_type_primary"
            >
              master
            </span>
            <span
              className="icon-plus__text icon-plus__text_type_secondary"
            >
              0cab04f
            </span>
          </div>
          <div
            className="icon-plus build-card__user"
          >
            <div
              className="icon icon_type_user icon_size_xs"
            />
            <span
              className="icon-plus__text icon-plus__text_type_primary"
            >
              Ilya Chudin
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="build-card build-card_view_default build-card_type_success build-list__item"
    onClick={[Function]}
  >
    <div
      className="build-card__status"
    />
    <div
      className="build-card__container"
    >
      <div>
        <div
          className="build-card__header"
        >
          <div
            className="build-card__number"
          >
            #
            36
          </div>
          <div
            className="build-card__title"
          >
            test update
          </div>
        </div>
        <div
          className="build-card__info"
        >
          <div
            className="icon-plus build-card__commit"
          >
            <div
              className="icon icon_type_commit icon_size_xs"
            />
            <span
              className="icon-plus__text icon-plus__text_type_primary"
            >
              master
            </span>
            <span
              className="icon-plus__text icon-plus__text_type_secondary"
            >
              0cab04f
            </span>
          </div>
          <div
            className="icon-plus build-card__user"
          >
            <div
              className="icon icon_type_user icon_size_xs"
            />
            <span
              className="icon-plus__text icon-plus__text_type_primary"
            >
              Ilya Chudin
            </span>
          </div>
        </div>
      </div>
      <div
        className="build-card__time"
      >
        <div
          className="icon-plus build-card__date"
        >
          <div
            className="icon icon_type_calendar icon_size_xs"
          />
          <span
            className="icon-plus__text icon-plus__text_type_secondary"
          >
            4 апр. 08:32
          </span>
        </div>
        <div
          className="icon-plus build-card__duration"
        >
          <div
            className="icon icon_type_stopwatch icon_size_xs"
          />
          <span
            className="icon-plus__text icon-plus__text_type_secondary"
          >
            39 м 35 сек
          </span>
        </div>
      </div>
    </div>
  </div>
  <button
    className="button button_size_m button_view_default build-list__button"
    onClick={[Function]}
    type="button"
  >
    <span
      className="button__text"
    >
      Show more
    </span>
  </button>
</div>
`;

exports[`BuildList component should renders correctly 3`] = `
<div
  className="build-list"
  data-testid="build-list"
>
  <div
    className="build-card build-card_view_default build-card_type_pending build-list__item"
    onClick={[Function]}
  >
    <div
      className="build-card__status"
    />
    <div
      className="build-card__container"
    >
      <div>
        <div
          className="build-card__header"
        >
          <div
            className="build-card__number"
          >
            #
            37
          </div>
          <div
            className="build-card__title"
          >
            test update
          </div>
        </div>
        <div
          className="build-card__info"
        >
          <div
            className="icon-plus build-card__commit"
          >
            <div
              className="icon icon_type_commit icon_size_xs"
            />
            <span
              className="icon-plus__text icon-plus__text_type_primary"
            >
              master
            </span>
            <span
              className="icon-plus__text icon-plus__text_type_secondary"
            >
              0cab04f
            </span>
          </div>
          <div
            className="icon-plus build-card__user"
          >
            <div
              className="icon icon_type_user icon_size_xs"
            />
            <span
              className="icon-plus__text icon-plus__text_type_primary"
            >
              Ilya Chudin
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="build-card build-card_view_default build-card_type_success build-list__item"
    onClick={[Function]}
  >
    <div
      className="build-card__status"
    />
    <div
      className="build-card__container"
    >
      <div>
        <div
          className="build-card__header"
        >
          <div
            className="build-card__number"
          >
            #
            36
          </div>
          <div
            className="build-card__title"
          >
            test update
          </div>
        </div>
        <div
          className="build-card__info"
        >
          <div
            className="icon-plus build-card__commit"
          >
            <div
              className="icon icon_type_commit icon_size_xs"
            />
            <span
              className="icon-plus__text icon-plus__text_type_primary"
            >
              master
            </span>
            <span
              className="icon-plus__text icon-plus__text_type_secondary"
            >
              0cab04f
            </span>
          </div>
          <div
            className="icon-plus build-card__user"
          >
            <div
              className="icon icon_type_user icon_size_xs"
            />
            <span
              className="icon-plus__text icon-plus__text_type_primary"
            >
              Ilya Chudin
            </span>
          </div>
        </div>
      </div>
      <div
        className="build-card__time"
      >
        <div
          className="icon-plus build-card__date"
        >
          <div
            className="icon icon_type_calendar icon_size_xs"
          />
          <span
            className="icon-plus__text icon-plus__text_type_secondary"
          >
            4 апр. 08:32
          </span>
        </div>
        <div
          className="icon-plus build-card__duration"
        >
          <div
            className="icon icon_type_stopwatch icon_size_xs"
          />
          <span
            className="icon-plus__text icon-plus__text_type_secondary"
          >
            39 м 35 сек
          </span>
        </div>
      </div>
    </div>
  </div>
  <div
    className="loading"
  >
    <div
      className="loading__container"
    >
      <div
        className="loading__bar"
      />
      <div
        className="loading__bar"
      />
      <div
        className="loading__bar"
      />
    </div>
  </div>
</div>
`;

exports[`BuildList component should renders correctly 4`] = `
<div
  className="build-list"
  data-testid="build-list"
>
  <div
    className="build-card build-card_view_default build-card_type_pending build-list__item"
    onClick={[Function]}
  >
    <div
      className="build-card__status"
    />
    <div
      className="build-card__container"
    >
      <div>
        <div
          className="build-card__header"
        >
          <div
            className="build-card__number"
          >
            #
            37
          </div>
          <div
            className="build-card__title"
          >
            test update
          </div>
        </div>
        <div
          className="build-card__info"
        >
          <div
            className="icon-plus build-card__commit"
          >
            <div
              className="icon icon_type_commit icon_size_xs"
            />
            <span
              className="icon-plus__text icon-plus__text_type_primary"
            >
              master
            </span>
            <span
              className="icon-plus__text icon-plus__text_type_secondary"
            >
              0cab04f
            </span>
          </div>
          <div
            className="icon-plus build-card__user"
          >
            <div
              className="icon icon_type_user icon_size_xs"
            />
            <span
              className="icon-plus__text icon-plus__text_type_primary"
            >
              Ilya Chudin
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="build-card build-card_view_default build-card_type_success build-list__item"
    onClick={[Function]}
  >
    <div
      className="build-card__status"
    />
    <div
      className="build-card__container"
    >
      <div>
        <div
          className="build-card__header"
        >
          <div
            className="build-card__number"
          >
            #
            36
          </div>
          <div
            className="build-card__title"
          >
            test update
          </div>
        </div>
        <div
          className="build-card__info"
        >
          <div
            className="icon-plus build-card__commit"
          >
            <div
              className="icon icon_type_commit icon_size_xs"
            />
            <span
              className="icon-plus__text icon-plus__text_type_primary"
            >
              master
            </span>
            <span
              className="icon-plus__text icon-plus__text_type_secondary"
            >
              0cab04f
            </span>
          </div>
          <div
            className="icon-plus build-card__user"
          >
            <div
              className="icon icon_type_user icon_size_xs"
            />
            <span
              className="icon-plus__text icon-plus__text_type_primary"
            >
              Ilya Chudin
            </span>
          </div>
        </div>
      </div>
      <div
        className="build-card__time"
      >
        <div
          className="icon-plus build-card__date"
        >
          <div
            className="icon icon_type_calendar icon_size_xs"
          />
          <span
            className="icon-plus__text icon-plus__text_type_secondary"
          >
            4 апр. 08:32
          </span>
        </div>
        <div
          className="icon-plus build-card__duration"
        >
          <div
            className="icon icon_type_stopwatch icon_size_xs"
          />
          <span
            className="icon-plus__text icon-plus__text_type_secondary"
          >
            39 м 35 сек
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
